### DISCUSSION TITLE
haptic feedback on iPhone

### DISCUSSION THREAD

**Discussion by Naozumi520** [ppy/osu#22916] [Ideas] [answered]
I think this can be a great starting point for mobile optimisation. Apple’s haptic engine provides low-latency haptic feedback. I think we can use it for the gameui/gameplay feedback, this can make user feels that the game is well-optimised for a phone, and not just a pure desktop game.

**Comment by peppy** [comment, marked as answer]
Definitely required for iOS, yep.
